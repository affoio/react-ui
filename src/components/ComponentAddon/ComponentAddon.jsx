import styled from 'styled-components';
import { oneOf, number } from 'prop-types';
import { positionEnum } from '../../enums/positionEnum';
import { getMultipliedIndent } from '../../theme/selectors';

const getAddonMargin = ({ position, marginSize, theme }) => {
  const indent = getMultipliedIndent(theme, marginSize);

  if (position === positionEnum.right) {
    return `
      margin-left: ${indent};
    `;
  }

  return `
      margin-right: ${indent};
    `;
};

const propTypes = {
  position: oneOf([positionEnum.left, positionEnum.right]),
  marginSize: number,
};

const ComponentAddon = styled.div.attrs(props => ({
  position: props.position || positionEnum.left,
  marginSize: props.marginSize || 1,
}))`
  display: flex;
  align-items: center;
  justify-content: center;
  ${getAddonMargin}
`;

ComponentAddon.propTypes = propTypes;

export default ComponentAddon;
