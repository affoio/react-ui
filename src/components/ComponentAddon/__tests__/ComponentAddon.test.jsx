import React from 'react';
import { render } from 'enzyme';
import ComponentAddon from '../ComponentAddon';

describe('<ComponentAddon>', () => {
  it('renders correctly', () => {
    const tree = render(<ComponentAddon />);

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with position right', () => {
    const tree = render(<ComponentAddon position="right" />);

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with marginSize', () => {
    const marginSizeList = [1, 2, 4];
    marginSizeList.forEach(size => {
      const tree = render(<ComponentAddon marginSize={size} />);

      expect(tree).toMatchSnapshot();
    });
  });
});
