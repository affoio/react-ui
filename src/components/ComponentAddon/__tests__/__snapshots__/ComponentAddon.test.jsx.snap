// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComponentAddon> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 4px;
}

<div
  class="c0"
/>
`;

exports[`<ComponentAddon> renders correctly with marginSize 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 4px;
}

<div
  class="c0"
/>
`;

exports[`<ComponentAddon> renders correctly with marginSize 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8px;
}

<div
  class="c0"
/>
`;

exports[`<ComponentAddon> renders correctly with marginSize 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 16px;
}

<div
  class="c0"
/>
`;

exports[`<ComponentAddon> renders correctly with position right 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 4px;
}

<div
  class="c0"
/>
`;
